src/app/api/profiles/[id]/state/route.ts

```diff
@@ -20,6 +19,17 @@ export async function PATCH(
   try {
     const userDataToUpdate: PublishingStateData = await request.json()
     const updatedProfileState = await updateUserData(id, userDataToUpdate)
+    const userData = await getUserById(id)
+    if (userData) {
```

Comment:
**nit:** No need for this logic - if you implement my first comment, let's just do:

```ts
  const updatedProfile = await updateUserData(id, userDataToUpdate)

if (updatedProfile.state === PublishingState.APPROVED) {
     sendProfileApprovalEmail(updatedProfile.email)
}
```

Rejecting should be done in other place (`api/profiles/[id]/reject/route.ts`). We should send reject reason upon rejecting profile